tests/src/lib.rs

```diff
@@ -37,6 +46,17 @@ mod tests {
         assert_eq!(token.allowance(BOB, ALI), 0);
     }
 
+    #[test]
+    #[should_panic]
+    fn test_approve_too_much() {
```

Comment:
This test fails at `token.transfer_from` not at `token.approve`. You can remove this test, as approve never panics.